Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/organizer/poster/QrCodePosterFragment.kt

```diff
@@ -0,0 +1,156 @@
+package de.rki.coronawarnapp.ui.eventregistration.organizer.poster
+
+import android.os.Bundle
+import android.print.PrintAttributes
+import android.print.PrintManager
+import android.util.TypedValue
+import android.view.View
+import android.view.ViewTreeObserver.OnGlobalLayoutListener
+import android.widget.Toast
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.content.getSystemService
+import androidx.core.widget.TextViewCompat
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.navArgs
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.databinding.QrCodePosterFragmentBinding
+import de.rki.coronawarnapp.exception.ExceptionCategory
+import de.rki.coronawarnapp.exception.reporting.report
+import de.rki.coronawarnapp.server.protocols.internal.pt.QrCodePosterTemplate.QRCodePosterTemplateAndroid.QRCodeTextBoxAndroid
+import de.rki.coronawarnapp.ui.color.parseColor
+import de.rki.coronawarnapp.ui.print.PrintingAdapter
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.files.FileSharing
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModelsAssisted
+import timber.log.Timber
+import java.io.File
+import javax.inject.Inject
+
+class QrCodePosterFragment : Fragment(R.layout.qr_code_poster_fragment), AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+
+    private val args by navArgs<QrCodePosterFragmentArgs>()
+    private val viewModel: QrCodePosterViewModel by cwaViewModelsAssisted(
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as QrCodePosterViewModel.Factory
+            factory.create(args.traceLocationId)
+        }
+    )
+
+    private val binding: QrCodePosterFragmentBinding by viewBindingLazy()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        with(binding) {
+            toolbar.setNavigationOnClickListener { popBackStack() }
+            viewModel.poster.observe(viewLifecycleOwner) { poster ->
+                bindPoster(poster)
+                // Avoid creating blank PDF
+                if (poster.hasImages()) onPosterDrawn()
+            }
+        }
+
+        viewModel.sharingIntent.observe(viewLifecycleOwner) {
+            onShareIntent(it)
+        }
+    }
+
+    private fun QrCodePosterFragmentBinding.bindPoster(poster: Poster) {
+        Timber.d("poster=$poster")
+        progressBar.hide()
+
+        val template = poster.template ?: return // Exit early
```

Comment:
Should we show the user that something went wrong?

Reply:
We do

Reply:
Please test it ?